src/run/service-factory/ServiceSpecRepository.ts

```diff
@@ -0,0 +1,180 @@
+import { existsSync, readFileSync, rmSync, writeFileSync } from "fs";
+import { addToStage, checkoutExistingBranch, checkoutNewBranch, cloneRepo, commit, push } from "../../helpers/git.js";
+import NewServiceSpec from "../../model/NewServiceSpec.js";
+import { join } from "path";
+import yaml from "yaml";
+
+/**
+ * A Repository for Service Specifications.
+ */
+export default class ServiceSpecRepository {
+
+    // eslint-disable-next-line no-useless-constructor
+    constructor (
+        private readonly localDirectory: string,
+        private readonly gitHubOrganisationName: string,
+        private readonly repositoryName: string,
+        private readonly branchName: string,
+        private readonly serviceSpecPath: string,
+        private readonly isPermanent: boolean = false
+    ) { }
+
+    /**
+     * Creates a new Service Specification Repository. When CHS_DEV_NEW_SERVICE_LOCAL_DIRECTORY
+     * is set uses this as the local directory otherwise uses the supplied local directory
+     * @param localDirectory Directory to clone the repository/use for the repository
+     * @param gitHubOrganisationName Name of the GitHub Organisation
+     * @param repositoryName Name of the GitHub Repository
+     * @param branchName Name of the branch to use
+     * @param serviceSpecPath Path to the service specifications within the repository
+     * @returns New Service Specification Repository
+     */
+    static create (
+        localDirectory: string,
+        gitHubOrganisationName: string,
+        repositoryName: string,
+        branchName: string,
+        serviceSpecPath: string
+    ) {
+        if (typeof process.env.CHS_DEV_NEW_SERVICE_LOCAL_DIRECTORY !== "undefined") {
```

Comment:
We could do this to avoid repetition.
`const envServiceDirectory = process.env.CHS_DEV_NEW_SERVICE_LOCAL_DIRECTORY;
 const serviceDirectory = envServiceDirectory || localDirectory;
 return new ServiceSpecRepository(
      serviceDirectory,
      gitHubOrganisationName,
      repositoryName,
      branchName,
      serviceSpecPath,
      Boolean(envServiceDirectory)
    );
`

Reply:
Nice Good suggestion 👍🏼